content/learning-paths/servers-and-cloud-computing/glibc-with-lse/mongo_benchmark.md

```diff
@@ -0,0 +1,140 @@
+---
+# User change
+title: "Benchmark MongoDB with YCSB"
+weight: 4 # 1 is first, 2 is second, etc.
+
+# Do not modify these elements
+layout: "learningpathall"
+---
+
+YCSB (Yahoo! Cloud Serving Benchmark) is a widely adopted open-source benchmarking tool designed to evaluate the performance of cloud-based or distributed data-serving systems. It was developed by Yahoo! and is now maintained by the open-source community.
+
+The primary goal of YCSB is to simulate different types of workloads commonly encountered in real-world cloud applications, such as read-heavy or write-heavy workloads. It provides a standardized framework for testing and comparing the performance of various data-serving systems, including databases, key-value stores, and distributed storage systems.
+
+YCSB supports a variety of popular data-serving systems, including Apache Cassandra, MongoDB, Redis, HBase, Amazon DynamoDB, and more. It provides a set of workload scenarios that can be customized to simulate specific application patterns and data access patterns.
+
+Using YCSB, you can measure key performance metrics like throughput, latency, and scalability of the target data-serving system under different workloads. This helps in evaluating the system's suitability for specific use cases, comparing different systems, and identifying performance bottlenecks or areas for optimization.
+
+YCSB is a command-line tool that provides a simple and extensible framework for benchmarking. It allows users to define their own workloads, extend it for new systems, and customize parameters such as the data distribution, request rate, and operation mix.
+
+Overall, YCSB has become a standard benchmarking tool in the cloud and distributed systems community, facilitating performance evaluations and enabling fair comparisons between various data-serving solutions.
+
+You are now ready to benchmark MongoDB with YCSB on your Arm server!
+
+## YCSB Setup
+Setup the YCSB on a benchmark machine with __JAVA__:
+```console
+cd ~
+wget -c https://github.com/brianfrankcooper/YCSB/releases/download/0.17.0/ycsb-0.17.0.tar.gz
+tar xfvz ycsb-0.17.0.tar.gz
+```
+
+Create a workload file with the following content:  
+```
+vi ~/ycsb-0.17.0/workloads/iworkload
+
+recordcount=1000  
+operationcount=1000  
+workload=site.ycsb.workloads.CoreWorkload  
+readallfields=true  
+readproportion=0.2  
+updateproportion=0.3  
+scanproportion=0.3  
+insertproportion=0  
+readmodifywriteproportion=0.2  
+requestdistribution=zipfian  
+
+```
+
+## YCSB Run
+ 1. To run YCSB, you need following the `load` command first: 
+    ```console
+    ~/ycsb-0.17.0/bin/ycsb.sh load mongodb -s -P ~/ycsb-0.17.0/workloads/iworkload -p recordcount=10000000 -threads 256 -p mongodb.url="mongodb://${mongo_ip}:${mongo_port}/mymongodb"
```

Comment:
I installed java:
sudo apt-get install -y openjdk-17-jre

java --version
openjdk 17.0.9 2023-10-17
OpenJDK Runtime Environment (build 17.0.9+9-Ubuntu-120.04)
OpenJDK 64-Bit Server VM (build 17.0.9+9-Ubuntu-120.04, mixed mode, sharing)

Then ran into the following error while running the YCSB workload:

Command line: -load -db site.ycsb.db.MongoDbClient -s -P /home/ubuntu/ycsb-0.17.0/workloads/iworkload -p recordcount=10000000 -threads 256 -p mongodb.url=mongodb://44.212.61.237:76719/mymongodb
YCSB Client 0.17.0

Loading workload...
java.lang.ClassNotFoundException: site.ycsb.workloads.CoreWorkload
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
        at site.ycsb.Client.getWorkload(Client.java:517)
        at site.ycsb.Client.main(Client.java:302)
java.lang.ClassNotFoundException: site.ycsb.workloads.CoreWorkload
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
        at site.ycsb.Client.getWorkload(Client.java:517)
        at site.ycsb.Client.main(Client.java:302)

Reply:
I haven't encountered the issue you mentioned.
You may try java-11.
sudo apt-get install -y openjdk-11-jdk